### DISCUSSION TITLE
node-chatgpt-api for Android

### DISCUSSION THREAD

**Discussion by araea** [viopsa233/AI-testers-Forum#8] [技术交流]
<html>
<body>
<div class="lake-content" typography="classic"><details class="lake-collapse"><summary id="u79cbb527"><strong><span class="ne-text" style="color: rgb(31, 35, 40)">前提:</span></strong></summary><ol class="ne-ol" style="margin: 0; padding-left: 23px"><li id="uf0111fce" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">Android手机 不需要root</span></li><li id="u53f1ee8b" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">魔法 自行解决 提示:github</span></li><li id="ucab8d735" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">微软账号 已通过new bing chat候选</span></li></ol></details><details class="lake-collapse"><summary id="u1c310453"><strong><span class="ne-text" style="color: rgb(31, 35, 40)">步骤:</span></strong></summary><ol class="ne-ol" style="margin: 0; padding-left: 23px"><li id="u677e2390" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">安装termux<br /></span><a href="https://github.com/termux/termux-app/releases" data-href="https://github.com/termux/termux-app/releases" target="_blank" class="ne-link"><span class="ne-text">https://github.com/termux/termux-app/releases</span></a></li><li id="ub357a8ef" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">先打开魔法，然后在termux中依次输入（复制粘贴用剪贴板)</span></li></ol><p id="u5923ec30" class="ne-p" style="margin: 0; padding: 0; min-height: 24px"><code class="ne-code" style="font-family: SFMono-Regular, Consolas, Liberation Mono, Menlo, Courier, monospace; background-color: rgba(0, 0, 0, 0.06); border: 1px solid rgba(0, 0, 0, 0.08); border-radius: 2px; padding: 0px 2px"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">apt update &amp;&amp; apt upgrade</span></code><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px"><br /></span><code class="ne-code" style="font-family: SFMono-Regular, Consolas, Liberation Mono, Menlo, Courier, monospace; background-color: rgba(0, 0, 0, 0.06); border: 1px solid rgba(0, 0, 0, 0.08); border-radius: 2px; padding: 0px 2px"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">pkg install nodejs</span></code><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px"><br /></span><code class="ne-code" style="font-family: SFMono-Regular, Consolas, Liberation Mono, Menlo, Courier, monospace; background-color: rgba(0, 0, 0, 0.06); border: 1px solid rgba(0, 0, 0, 0.08); border-radius: 2px; padding: 0px 2px"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">pkg install git</span></code><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px"><br /></span><code class="ne-code" style="font-family: SFMono-Regular, Consolas, Liberation Mono, Menlo, Courier, monospace; background-color: rgba(0, 0, 0, 0.06); border: 1px solid rgba(0, 0, 0, 0.08); border-radius: 2px; padding: 0px 2px"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">mkdir bing</span></code><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px"><br /></span><code class="ne-code" style="font-family: SFMono-Regular, Consolas, Liberation Mono, Menlo, Courier, monospace; background-color: rgba(0, 0, 0, 0.06); border: 1px solid rgba(0, 0, 0, 0.08); border-radius: 2px; padding: 0px 2px"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">cd bing</span></code><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px"><br /></span><code class="ne-code" style="font-family: SFMono-Regular, Consolas, Liberation Mono, Menlo, Courier, monospace; background-color: rgba(0, 0, 0, 0.06); border: 1px solid rgba(0, 0, 0, 0.08); border-radius: 2px; padding: 0px 2px"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">git clone </span><a href="https://github.com/waylaidwanderer/node-chatgpt-api.git" data-href="https://github.com/waylaidwanderer/node-chatgpt-api.git" target="_blank" class="ne-link"><span class="ne-text">https://github.com/waylaidwanderer/node-chatgpt-api.git</span></a></code><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px"><br /></span><code class="ne-code" style="font-family: SFMono-Regular, Consolas, Liberation Mono, Menlo, Courier, monospace; background-color: rgba(0, 0, 0, 0.06); border: 1px solid rgba(0, 0, 0, 0.08); border-radius: 2px; padding: 0px 2px"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">cd node-chatgpt-api</span></code><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px"><br /></span><code class="ne-code" style="font-family: SFMono-Regular, Consolas, Liberation Mono, Menlo, Courier, monospace; background-color: rgba(0, 0, 0, 0.06); border: 1px solid rgba(0, 0, 0, 0.08); border-radius: 2px; padding: 0px 2px"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">npm install</span></code><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px"><br /></span><code class="ne-code" style="font-family: SFMono-Regular, Consolas, Liberation Mono, Menlo, Courier, monospace; background-color: rgba(0, 0, 0, 0.06); border: 1px solid rgba(0, 0, 0, 0.08); border-radius: 2px; padding: 0px 2px"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">cp settings.example.js settings.js</span></code><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px"><br /></span><code class="ne-code" style="font-family: SFMono-Regular, Consolas, Liberation Mono, Menlo, Courier, monospace; background-color: rgba(0, 0, 0, 0.06); border: 1px solid rgba(0, 0, 0, 0.08); border-radius: 2px; padding: 0px 2px"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">pkg install nano</span></code><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px"><br /></span><code class="ne-code" style="font-family: SFMono-Regular, Consolas, Liberation Mono, Menlo, Courier, monospace; background-color: rgba(0, 0, 0, 0.06); border: 1px solid rgba(0, 0, 0, 0.08); border-radius: 2px; padding: 0px 2px"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">nano settings.js</span></code></p><ul class="ne-ul" style="margin: 0; padding-left: 23px"><li id="u30b3496b" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">此时为文本编辑模式</span></li><li id="u11a06825" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">找到“bingAiClient”下的“cookies”(如果你不知道怎么获取</span><a href="https://www.bing.com/" data-href="https://www.bing.com/" target="_blank" class="ne-link"><span class="ne-text">https://www.bing.com/</span></a><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">的cookies,请看下面的补充)</span></li><li id="u4be7a166" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">将你的cookies填入单引号内</span></li><li id="u82faf8ae" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">紧接着，拉到最下面，把最后一个//注释符去掉</span></li><li id="ub36d0025" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">完成之后,点击底部Ctrl,再用输入法点一下S保存文件</span></li><li id="u4ebcadda" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">再Ctrl X退出文本编辑</span></li><li id="u47f39c8e" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">输入</span><code class="ne-code" style="font-family: SFMono-Regular, Consolas, Liberation Mono, Menlo, Courier, monospace; background-color: rgba(0, 0, 0, 0.06); border: 1px solid rgba(0, 0, 0, 0.08); border-radius: 2px; padding: 0px 2px"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">npm run cli</span></code></li><li id="ud380fffb" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">现在,你应该已经成功了</span></li></ul><ol start="3" class="ne-ol" style="margin: 0; padding-left: 23px"><li id="u17df33c3" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">下一次打开termux 依次输入<br /></span><code class="ne-code" style="font-family: SFMono-Regular, Consolas, Liberation Mono, Menlo, Courier, monospace; background-color: rgba(0, 0, 0, 0.06); border: 1px solid rgba(0, 0, 0, 0.08); border-radius: 2px; padding: 0px 2px"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">cd bing/node-chatgpt-api</span></code><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px"><br /></span><code class="ne-code" style="font-family: SFMono-Regular, Consolas, Liberation Mono, Menlo, Courier, monospace; background-color: rgba(0, 0, 0, 0.06); border: 1px solid rgba(0, 0, 0, 0.08); border-radius: 2px; padding: 0px 2px"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">npm run cli</span></code></li></ol></details><details class="lake-collapse"><summary id="u5a571dd6"><strong><span class="ne-text" style="color: rgb(31, 35, 40)">补充:</span></strong></summary><ol class="ne-ol" style="margin: 0; padding-left: 23px"><li id="uc0efb0a8" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">想要获取bing.com的全部cookies 这里是一个可能的办法</span></li></ol><ul class="ne-ul" style="margin: 0; padding-left: 23px"><li id="u4b58e875" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">在电脑打开Edge 打开Microsoft Edge加载项 搜索并获取Tampermonkey<br /></span><a href="https://microsoftedge.microsoft.com/addons/detail/tampermonkey/iikmkjmpaadaobahmlepeloendndfphd" data-href="https://microsoftedge.microsoft.com/addons/detail/tampermonkey/iikmkjmpaadaobahmlepeloendndfphd" target="_blank" class="ne-link"><span class="ne-text">https://microsoftedge.microsoft.com/addons/detail/tampermonkey/iikmkjmpaadaobahmlepeloendndfph</span></a></li><li id="u42d1fb9b" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">在Greasy Fork网站上搜索并安装“一键获取网站cookie”<br /></span><a href="https://greasyfork.org/zh-CN/scripts/455411-getthecookie" data-href="https://greasyfork.org/zh-CN/scripts/455411-getthecookie" target="_blank" class="ne-link"><span class="ne-text">https://greasyfork.org/zh-CN/scripts/455411-getthecookie</span></a></li><li id="u6b2ed1a7" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">打开</span><a href="https://www.bing.com/" data-href="https://www.bing.com/" target="_blank" class="ne-link"><span class="ne-text">https://www.bing.com/</span></a><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px"> 点击左下角CK,Cookies到手</span></li></ul><ol class="ne-ol" style="margin: 0; padding-left: 23px"><li id="u411a85da" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">默认为平衡模式,更改为有创造力模式方法:</span></li></ol><ul class="ne-ul" style="margin: 0; padding-left: 23px"><li id="u0e88367f" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">打开termux<br /></span><code class="ne-code" style="font-family: SFMono-Regular, Consolas, Liberation Mono, Menlo, Courier, monospace; background-color: rgba(0, 0, 0, 0.06); border: 1px solid rgba(0, 0, 0, 0.08); border-radius: 2px; padding: 0px 2px"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">cd bing/node-chatgpt-api/src</span></code><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px"><br /></span><code class="ne-code" style="font-family: SFMono-Regular, Consolas, Liberation Mono, Menlo, Courier, monospace; background-color: rgba(0, 0, 0, 0.06); border: 1px solid rgba(0, 0, 0, 0.08); border-radius: 2px; padding: 0px 2px"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">nano BingAIClient.js</span></code></li><li id="u1b872483" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">找到“toneStyle” 根据注释修改即可 修改完毕后</span></li><li id="u1b57d2a8" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">Ctrl S保存</span></li><li id="u3a3dd07c" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">Ctrl X退出</span></li></ul><ol class="ne-ol" style="margin: 0; padding-left: 23px"><li id="u72d17f0f" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">默认为Sydney人工智能助手,更改设定(提示注入)方法:</span></li></ol><ul class="ne-ul" style="margin: 0; padding-left: 23px"><li id="u7422b0d2" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">打开termux<br /></span><code class="ne-code" style="font-family: SFMono-Regular, Consolas, Liberation Mono, Menlo, Courier, monospace; background-color: rgba(0, 0, 0, 0.06); border: 1px solid rgba(0, 0, 0, 0.08); border-radius: 2px; padding: 0px 2px"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">cd bing/node-chatgpt-api/src</span></code><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px"><br /></span><code class="ne-code" style="font-family: SFMono-Regular, Consolas, Liberation Mono, Menlo, Courier, monospace; background-color: rgba(0, 0, 0, 0.06); border: 1px solid rgba(0, 0, 0, 0.08); border-radius: 2px; padding: 0px 2px"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">nano BingAIClient.js</span></code></li><li id="u3ae2b811" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">找到 </span><em><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">text: systemMessage || &quot;You're an AI assistant named Sydney. Instead of introducing yourself as Bing,</span></em></li><li id="uf276945e" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">更改双引号中内容即可</span></li></ul><ol class="ne-ol" style="margin: 0; padding-left: 23px"><li id="u81aa97c7" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">聊天记录会保存在“cache.json”中,查看方法:</span></li></ol><ul class="ne-ul" style="margin: 0; padding-left: 23px"><li id="u6e879aab" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">打开termux<br /></span><code class="ne-code" style="font-family: SFMono-Regular, Consolas, Liberation Mono, Menlo, Courier, monospace; background-color: rgba(0, 0, 0, 0.06); border: 1px solid rgba(0, 0, 0, 0.08); border-radius: 2px; padding: 0px 2px"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">cd bing/node-chatgpt-api</span></code><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px"><br /></span><code class="ne-code" style="font-family: SFMono-Regular, Consolas, Liberation Mono, Menlo, Courier, monospace; background-color: rgba(0, 0, 0, 0.06); border: 1px solid rgba(0, 0, 0, 0.08); border-radius: 2px; padding: 0px 2px"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">nano cache.json</span></code></li></ul><ol class="ne-ol" style="margin: 0; padding-left: 23px"><li id="u67adecd7" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">如果你知道你魔法的端口xxxx,可以在“settings.js”的“cookies”下面的“proxy”填写</span></li></ol><ul class="ne-ul" style="margin: 0; padding-left: 23px"><li id="uf0a25ebf" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">proxy: '</span><span class="ne-text" style="color: var(--color-accent-fg); font-size: 14px">http://127.0.0.1:xxxx</span><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">'</span></li></ul><ol start="2" class="ne-ol" style="margin: 0; padding-left: 23px"><li id="u8c6b3828" data-lake-index-type="0"><span class="ne-text" style="color: rgb(31, 35, 40); font-size: 14px">node-chatgpt-api for Windows -&gt; 第一个参考链接</span></li></ol></details><details class="lake-collapse"><summary id="u7c5d57a4"><strong><span class="ne-text" style="color: rgb(31, 35, 40)">参考链接:</span></strong></summary><ol class="ne-ol" style="margin: 0; padding-left: 23px"><li id="u54e0b44c" data-lake-index-type="0"><a href="https://github.com/iouzzrj/node-chatgpt-apiTutorial/blob/master/bingTutorial.md" data-href="https://github.com/iouzzrj/node-chatgpt-apiTutorial/blob/master/bingTutorial.md" target="_blank" class="ne-link"><span class="ne-text">https://github.com/iouzzrj/node-chatgpt-apiTutorial/blob/master/bingTutorial.md</span></a></li><li id="ub9dfe154" data-lake-index-type="0"><a href="https://github.com/waylaidwanderer/node-chatgpt-api" data-href="https://github.com/waylaidwanderer/node-chatgpt-api" target="_blank" class="ne-link"><span class="ne-text">https://github.com/waylaidwanderer/node-chatgpt-api</span></a></li></ol></details>
</div>
</body>
</html>

**Comment by tzcnbo** [comment]
确实挺不错的，但是用着用着就会变成"你好，我是Sydney，你的AI助手"这句话怎么搞

**Reply by viopsa233** [reply]
node-chatgpt-api本身的BUG,没办法

**Comment by xuyufenfei** [comment]
![Screenshot_2023-07-13-23-07-21-548_com termux](https://github.com/viopsa233/AI-testers-Forum/assets/133896005/d87832e9-67d5-4147-9882-b6e470a7240b)
